sql/core/src/test/scala/org/apache/spark/sql/execution/BaseScriptTransformationSuite.scala

```diff
@@ -0,0 +1,390 @@
+          StructField("d", ByteType, false),
+          StructField("e", DecimalType(38, 18), true),
+          StructField("f", TimestampType, true),
+          StructField("g", DateType, true),
+          StructField("h", CalendarIntervalType, true),
+          StructField("i", ArrayType(IntegerType, false), true),
+          StructField("j", MapType(StringType, IntegerType, false), true),
+          StructField("k", new TestUDT.MyDenseVectorUDT, true),
+          StructField("l", new SimpleTupleUDT, true),
+          StructField("m", StructType(
+            Seq(StructField("a", IntegerType, false),
+              StructField("b", StringType, true))), false),
+          StructField("n", BinaryType, true),
+          StructField("o", BooleanType, false))))
+
+      // Can't support convert script output data to ArrayType/MapType/StructType now,
+      // return these column still as string.
+      // For UserDefinedType, if user defined deserialize method to support convert string
+      // to UserType like [[SimpleTupleUDT]], we can support convert to this UDT, else we
+      // will return null value as column.
+      checkAnswer(
+        df,
+        (child: SparkPlan) => createScriptTransformationExec(
+          input = Seq(
+            df.col("a").expr,
+            df.col("b").expr,
+            df.col("c").expr,
+            df.col("d").expr,
+            df.col("e").expr,
+            df.col("f").expr,
+            df.col("g").expr,
+            df.col("h").expr,
+            df.col("i").expr,
+            df.col("j").expr,
+            df.col("k").expr,
+            df.col("l").expr,
+            df.col("m").expr,
+            df.col("n").expr,
+            df.col("o").expr),
+          script = "cat",
+          output = Seq(
+            AttributeReference("a", IntegerType)(),
+            AttributeReference("b", StringType)(),
+            AttributeReference("c", DoubleType)(),
+            AttributeReference("d", ByteType)(),
+            AttributeReference("e", DecimalType(38, 18))(),
+            AttributeReference("f", TimestampType)(),
+            AttributeReference("g", DateType)(),
+            AttributeReference("h", CalendarIntervalType)(),
+            AttributeReference("i", StringType)(),
+            AttributeReference("j", StringType)(),
+            AttributeReference("k", StringType)(),
+            AttributeReference("l", new SimpleTupleUDT)(),
+            AttributeReference("m", StringType)(),
+            AttributeReference("n", BinaryType)(),
+            AttributeReference("o", BooleanType)()),
+          child = child,
+          ioschema = defaultIOSchema
+        ),
+        df.select(
+          'a, 'b, 'c, 'd, 'e, 'f, 'g, 'h,
+          'i.cast("string"),
+          'j.cast("string"),
+          'k.cast("string"),
+          'l, 'm.cast("string"), 'n, 'o).collect())
+    }
+  }
+
+  test("SPARK-32106: TRANSFORM shoud return null when return string incompatible(no serde)") {
```

Comment:
Could you move this tests into `SQLQueryTestSuite`?

Reply:
`shoud` -> `should`

Reply:
> Could you move this tests into `SQLQueryTestSuite`?

Have done,  the last sql is this. https://github.com/apache/spark/pull/29085#discussion_r456847499